### PR TITLE
Package caper.0.9

### PR DESCRIPTION
**Pull request by spaghetus**
# `caper.0.9`

Caper is a tool for understanding and processing "pcap expressions" (also known as tcpdump filters) which are used for network packet analysis.

---

* Homepage: https://gitlab.com/niksu/caper
* Source repo: git://gitlab.com/niksu/caper.git
* Bug tracker: https://gitlab.com/niksu/caper/issues

### CODE REVIEW

File: packages/caper/caper.0.9/opam [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+description: """
+Caper is a tool for understanding and processing "pcap expressions" (also known as tcpdump filters) which are used for network packet analysis.
+"""
+maintainer: "Willow Carlson-Huber <opam@w.wolo.dev>"
+authors: "https://gitlab.com/niksu/caper#contributors"
+license: "GPL-3.0-or-later"
+homepage: "https://gitlab.com/niksu/caper"
+bug-reports: "https://gitlab.com/niksu/caper/issues"
+dev-repo: "git://gitlab.com/niksu/caper.git"
+depends: ["ocaml" "ocamlbuild" "menhir" "angstrom" "conf-m4" "ocamlfind"]
```
Comment (mseri) [inline packages/caper/caper.0.9/opam:12]:
```suggestion
depends: [
  "ocaml" {>= "4.08"}
  "ocamlbuild"
  "menhir"
  "angstrom"
  "angstrom" {>= "0.16.0" & with-test}
  "conf-m4"
  "ocamlfind"
]
```
This should solve the current issues

File: packages/caper/caper.0.9/opam [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+authors: "https://gitlab.com/niksu/caper#contributors"
+license: "GPL-3.0-or-later"
+homepage: "https://gitlab.com/niksu/caper"
+bug-reports: "https://gitlab.com/niksu/caper/issues"
+dev-repo: "git://gitlab.com/niksu/caper.git"
+depends: [
+  "ocaml" {>= "4.08"}
+  "ocamlbuild"
+  "menhir"
+  "angstrom"
```
Comment (avsm) [inline packages/caper/caper.0.9/opam:16]:
```suggestion
  "angstrom" {>= "0.10.0"}
```

also a lower bound for `failure_msg` use

File: packages/caper/caper.0.9/opam [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+authors: "https://gitlab.com/niksu/caper#contributors"
+license: "GPL-3.0-or-later"
+homepage: "https://gitlab.com/niksu/caper"
+bug-reports: "https://gitlab.com/niksu/caper/issues"
+dev-repo: "git://gitlab.com/niksu/caper.git"
+depends: [
+  "ocaml" {>= "4.08"}
+  "ocamlbuild"
+  "menhir"
+  "angstrom" {>= "0.10.0"}
```
Comment (mseri) [inline packages/caper/caper.0.9/opam:16]:
```suggestion
  "angstrom" {>= "0.14.0"}
```

`~consume` in `parse_string` was added in this version

File: packages/caper/caper.0.9/opam [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+depends: [
+  "ocaml" {>= "4.08"}
+  "ocamlbuild"
+  "menhir"
+  "angstrom" {>= "0.14.0"}
+  "angstrom" {>= "0.16.0" & with-test}
+  "conf-m4"
+  "ocamlfind"
+]
+build: [
```
Comment (mseri) [inline packages/caper/caper.0.9/opam:21]:
```suggestion
available: arch != "x86_32" & arch != "arm32" & arch != "ppc64"
conflicts: [ "ocaml-option-bytecode-only" ]
build: [
```
due to
```
# File "pcap_to_bpf/util.ml", line 139, characters 14-24:
# 139 |   | [oct1] -> 0xff000000
#                     ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
```
and
```
# + ocamlfind ocamlopt -c -g -thread -package angstrom -package str -I english -I syntax -I pcap_to_bpf -I pcap_to_bpf/protocol -I pcap_to_bpf/optimizer -o english/english_syntax.cmx english/english_syntax.ml
# ocamlfind: Not supported in your configuration: ocamlopt
```

**Comment by mseri** [comment]
The code has a very long list of warnings with redundant or unused match cases, I'd recommend to have a look

**Comment by mseri** [comment]
Thanks!

@spaghetus please port these fixes to the opam files in the main branch of your repository

**Pull request merged**